### DISCUSSION TITLE
Breaking and major changes for future 0.7.0 release

### DISCUSSION THREAD

**Discussion by christianparpart** [contour-terminal/contour#1594] [Development]
This is the incomplete list of breaking changes we plan to get in for next big 0.7.0 release:

- cleanus:
  - [ ] replace ranges-v3 with std::ranges
  - [x] replace fmtlib with std::format
  - [x] drop very thing older than 22.04
- additions:
  - [x] #1584 
  - [ ] :skull: libunicode grapheme cluster segmentation PR finally merged :skull: 
- improvements:
  - improve search experience
  - improve IME on macOS (maybe make it configurable via shortcut)
- think big:
  - #1101 
  - #1010
  - popup VT extension?, get inspired by how MS Windows Terminal is doing it (recycle their VT seqs, if any) (see: https://github.com/microsoft/vscode/issues/155639)
  - Windows default terminal: set it to contour
- bugs without tickets yet?
  - mouse position calculation is wrong on DPI scaling != 1.0 ?
  - DPI scaling change breaks calculation (in case it worked before?), e.g. on monitor/screen change?
  - visual mode word wise select does select too much to the left at least (at least on Windows/Conhost)

**Comment by Yaraslaut** [comment]
Also i think we can switch to c++23
